player/configfiles.c

```diff
@@ -192,6 +192,14 @@ static char *mp_get_playback_resume_config_filename(struct MPContext *mpctx,
         conf = talloc_asprintf_append(conf, "%02X", md5[i]);
 
     if (!mpctx->cached_watch_later_configdir) {
+        char *wl_dir = mpctx->opts->watch_later_directory;
+        if (wl_dir && wl_dir[0]) {
+            mpctx->cached_watch_later_configdir = mp_get_user_path(NULL, mpctx->global, wl_dir);
+            mp_mkdirp(mpctx->cached_watch_later_configdir);
```

Comment:
Apologies if I'm misunderstanding something here, but my change doesn't seem to do that either:

    $ ls /tmp/mpv-watch-later
    ls: /tmp/mpv-watch-later: No such file or directory
    $ ./build/mpv test.mp3
    $ ls /tmp/mpv-watch-later
    ls: /tmp/mpv-watch-later: No such file or directory
    $ ./build/mpv --watch-later-directory=/tmp/mpv-watch-later test.mp3
    $ ls /tmp/mpv-watch-later
    503B8962F2A2439FC8A980533308C832